### PR TITLE
[MAINT] clean up makefiles and allow color in CLI in github workflows

### LINKED ISSUE #4314: clean up makefiles
It may be worth checking if all the "recipes" listed in the makefiles (in the root folder AND in the doc) can still be run and are useful / used.

For the makefile in the root folder, it may be useful to check if some of them cannot be transferred to `tox.ini`.

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4314

Changes proposed in this pull request:

- clean up makefiles to remove unused recipes
- transfer basic doc building from main makefile to tox.ini
- add recipe in doc/makefile to allow easy viewing of the built doc with `make view` or `make -D doc view`
- update tox config and CI workflows so that the tests and doc output shows color in CI reports

### CODE REVIEW

File: doc/Makefile [hunk 1]
```diff
@@ -158,21 +174,11 @@ doctest:
...
 	cp -r _build/html _build/nilearn ;
-	cp -r data _build/nilearn ;
-	cp nilearn.pdf _build/nilearn;
 	zip -r _build/nilearn.zip _build/nilearn
 
-pdf: latex
-	cd _build/latex ; make all-pdf ; pdfnup niearn.pdf
-	cp _build/latex/nilearn.pdf nilearn-simple.pdf
-	cp _build/latex/nilearn-nup.pdf nilearn.pdf
-	#cd _build/latex ; make all-pdf ; pdfnup nilearn.pdf
```
Comment (Remi-Gau) [inline doc/Makefile:174]: This assumes that we do not want to publish the doc as pdf beyond what a latex build can give us.

File: Makefile [hunk 2]
```diff
@@ -19,34 +17,4 @@ clean-so:
...
-trailing-spaces:
-	find . -name "*.py" | xargs perl -pi -e 's/[ \t]*$$//'
-
-cython:
-	find -name "*.pyx" | xargs $(CYTHON)
-
-ctags:
-	# make tags for symbol based navigation in emacs and vim
-	# Install with: sudo apt-get install exuberant-ctags
-	$(CTAGS) -R *
```
Comment (Remi-Gau) [inline Makefile:36]: ok maybe I should leave this for emacs users.

File: Makefile [hunk 3]
```diff
@@ -19,34 +17,4 @@ clean-so:
 clean-build:
 	rm -rf build
 
-clean-ctags:
-	rm -f tags
-
-clean: clean-build clean-pyc clean-so clean-ctags
-
-trailing-spaces:
-	find . -name "*.py" | xargs perl -pi -e 's/[ \t]*$$//'
```
Comment (Remi-Gau) [inline Makefile:28]: pre-commit should take care of this

File: Makefile [hunk 4]
```diff
@@ -19,34 +17,4 @@ clean-so:
...
-	# Install with: sudo apt-get install exuberant-ctags
-	$(CTAGS) -R *
-
-.PHONY : doc-plot
-doc-plot:
-	make -C doc html
-
-.PHONY : doc
-doc:
-	make -C doc html-noplot
```
Comment (Remi-Gau) [inline Makefile:44]: transferred to tox.ini that also takes care of the environment

File: Makefile [hunk 5]
```diff
@@ -19,34 +17,4 @@ clean-so:
...
-
-.PHONY : ci-doc
-ci-doc:
-	make -C doc ci-html-noplot
```
Comment (Remi-Gau) [inline Makefile:48]: not used in any of our CI build AFAICT

**Review by bthirion — APPROVED** [review]
LGTM. To be frank  the PR is cryptic to me, but the proof is in the result.

**Pull request merged**